src/test/java/com/thirdcc/webapp/web/rest/EventActivityResourceIT.java

```diff
@@ -275,25 +448,217 @@ public void updateNonExistingEventActivity() throws Exception {
     }
 
     @Test
-    @Transactional
+    public void updateEventActivity_WithEventEnded_ShouldThrow400() throws Exception {
+        event.setEndDate(Instant.now().minus(1, ChronoUnit.DAYS));
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeUpdate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithEventClosed_ShouldThrow400() throws Exception {
+        event.setStatus(EventStatus.CLOSED);
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeUpdate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithEventCancelled_ShouldThrow400() throws Exception {
+        event.setStatus(EventStatus.CANCELLED);
+        Event savedEvent = initEventDB();
+        EventActivity savedEventActivity = initEventActivityDB(savedEvent);
+
+        int databaseSizeBeforeUpdate = eventActivityRepository.findAll().size();
+
+        EventActivityDTO eventActivityDTO = createUpdateEventActivityDTO();
+        eventActivityDTO.setId(savedEventActivity.getId());
+
+        restEventActivityMockMvc.perform(put("/api/event-activities")
+            .contentType(TestUtil.APPLICATION_JSON_UTF8)
+            .content(TestUtil.convertObjectToJsonBytes(eventActivityDTO)))
+            .andExpect(status().isBadRequest());
+
+        List<EventActivity> eventActivityList = eventActivityRepository.findAll();
+        assertThat(eventActivityList).hasSize(databaseSizeBeforeUpdate);
+    }
+
+    @Test
+    public void updateEventActivity_WithEventEarlierThanToday_ShouldThrow400() throws Exception {
```

Comment:
this is test for event startDate earlier than today?
or test eventActivity startDate update to earlier than today?

Reply:
this is a typo
`updateEventActivity_WithEventEarlierThanToday_ShouldThrow400` -> `updateEventActivity_WithEventActivityStartDateEarlierThanToday_ShouldThrow400`

fixed in 7b0c8c424cf67f43c64cdf692fb94d36f7b858c5